### ISSUE TITLE
Cannot restore previous file version with enabled encryption #2

### ISSUE THREAD

**Issue by shhristov** [owncloud/core#23140]
### Steps to reproduce
1. Install ownCloud 9.0.0 (stable) (Ubuntu server 14.04, owncloud's repository, SQLite DB)
2. Enable Default encryption module.
3. Enable Server-side encryption with "Encrypt home storage option"
4. Logout/login
5. Upload random .docx file by using the web interface(I used this one: http://www.ijirset.com/IJIRSET_Paper_Template.docx)
6. Change the file by editing it locally on your PC.
7. Re-upload the file by using the web interface, overwriting the initially uploaded file.
8. By using the web interface revert to the previous version of the file.
9. Download the file.

NOTE: If you skip step 6, it works fine.

### Expected behaviour
Successful download.

### Actual behaviour
Download fails. The error from Firefox is: "/tmp/mozilla_john0/U4w2tJUE.docx.part could not be saved, because the source file could not be read."

### Server configuration
**Operating system**: Ubuntu 14.04.4 LTS

**Web server:** Apache/2.4.7 (Ubuntu)

**Database:** SQLite3

**PHP version:** PHP 5.5.9-1ubuntu4.14

**ownCloud version:** ownCloud 9.0.0 (stable)

**Updated from an older ownCloud or fresh install:** No

**Where did you install ownCloud from:** Official repository: deb http://download.owncloud.org/download/repositories/stable/xUbuntu_14.04/ /

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
root@owncloud:~# sudo -u www-data php /var/www/owncloud/occ app:list
Enabled:
  - activity: 2.2.1
  - comments: 0.2
  - dav: 0.1.5
  - encryption: 1.2.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - external
  - files_external
  - user_external
  - user_ldap

```

**The content of config/config.php:**

```
root@owncloud:~# sudo -u www-data php /var/www/owncloud/occ config:list system
{
    "system": {
        "updatechecker": false,
        "instanceid": "ocrqypq3qpwf",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "owncloud.programista.pro"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "http:\/\/owncloud.programista.pro",
        "dbtype": "sqlite3",
        "version": "9.0.0.19",
        "logtimezone": "UTC",
        "installed": true
    }
}

```

**Are you using external storage, if yes which one:** None

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** None

### Client configuration
**Browser:** Firefox 44.0.2

**Operating system:** Linux Mint 17.3 Rosa

#### Web server error log
```
root@owncloud:~# cat /var/log/apache2/error.log 
[Thu Mar 10 17:36:21.426239 2016] [mpm_event:notice] [pid 6952:tid 139648392947584] AH00489: Apache/2.4.7 (Ubuntu) configured -- resuming normal operations
[Thu Mar 10 17:36:21.426335 2016] [core:notice] [pid 6952:tid 139648392947584] AH00094: Command line: '/usr/sbin/apache2'
[Thu Mar 10 17:36:25.762979 2016] [mpm_event:notice] [pid 6952:tid 139648392947584] AH00491: caught SIGTERM, shutting down
[Thu Mar 10 17:36:26.811222 2016] [mpm_prefork:notice] [pid 8370] AH00163: Apache/2.4.7 (Ubuntu) configured -- resuming normal operations
[Thu Mar 10 17:36:26.811313 2016] [core:notice] [pid 8370] AH00094: Command line: '/usr/sbin/apache2'
[Thu Mar 10 17:36:28.031487 2016] [mpm_prefork:notice] [pid 8370] AH00169: caught SIGTERM, shutting down
[Thu Mar 10 17:36:29.176639 2016] [mpm_prefork:notice] [pid 8456] AH00163: Apache/2.4.7 (Ubuntu) PHP/5.5.9-1ubuntu4.14 configured -- resuming normal operations
[Thu Mar 10 17:36:29.176713 2016] [core:notice] [pid 8456] AH00094: Command line: '/usr/sbin/apache2'
[Thu Mar 10 17:36:35.277058 2016] [mpm_prefork:notice] [pid 8456] AH00171: Graceful restart requested, doing restart
[Thu Mar 10 17:36:35.329191 2016] [mpm_prefork:notice] [pid 8456] AH00163: Apache/2.4.7 (Ubuntu) PHP/5.5.9-1ubuntu4.14 configured -- resuming normal operations
[Thu Mar 10 17:36:35.329205 2016] [core:notice] [pid 8456] AH00094: Command line: '/usr/sbin/apache2'
[Thu Mar 10 17:36:40.533134 2016] [mpm_prefork:notice] [pid 8456] AH00169: caught SIGTERM, shutting down
[Thu Mar 10 17:36:41.604467 2016] [mpm_prefork:notice] [pid 8590] AH00163: Apache/2.4.7 (Ubuntu) PHP/5.5.9-1ubuntu4.14 configured -- resuming normal operations
[Thu Mar 10 17:36:41.604512 2016] [core:notice] [pid 8590] AH00094: Command line: '/usr/sbin/apache2'
[Thu Mar 10 17:36:47.756789 2016] [mpm_prefork:notice] [pid 8590] AH00171: Graceful restart requested, doing restart
[Thu Mar 10 17:36:47.808727 2016] [mpm_prefork:notice] [pid 8590] AH00163: Apache/2.4.7 (Ubuntu) PHP/5.5.9-1ubuntu4.14 configured -- resuming normal operations
[Thu Mar 10 17:36:47.808741 2016] [core:notice] [pid 8590] AH00094: Command line: '/usr/sbin/apache2'
[Thu Mar 10 17:37:34.128271 2016] [mpm_prefork:notice] [pid 8590] AH00171: Graceful restart requested, doing restart
[Thu Mar 10 17:37:34.173858 2016] [mpm_prefork:notice] [pid 8590] AH00163: Apache/2.4.7 (Ubuntu) PHP/5.5.9-1ubuntu4.14 configured -- resuming normal operations
[Thu Mar 10 17:37:34.173872 2016] [core:notice] [pid 8590] AH00094: Command line: '/usr/sbin/apache2'
[Thu Mar 10 17:38:12.253585 2016] [:error] [pid 8720] [client 10.194.141.100:45641] creating sqlite db
[Thu Mar 10 17:40:39.444864 2016] [authz_core:error] [pid 8720] [client 10.194.141.100:45682] AH01630: client denied by server configuration: /var/www/owncloud/data/.ocdata
[Thu Mar 10 17:41:19.387613 2016] [authz_core:error] [pid 8719] [client 10.194.141.100:45707] AH01630: client denied by server configuration: /var/www/owncloud/data/.ocdata
[Thu Mar 10 18:03:38.657262 2016] [authz_core:error] [pid 8719] [client 10.194.141.100:46063] AH01630: client denied by server configuration: /var/www/owncloud/data/.ocdata
[Thu Mar 10 18:25:43.966673 2016] [authz_core:error] [pid 8715] [client 10.194.141.100:46380] AH01630: client denied by server configuration: /var/www/owncloud/data/.ocdata
[Fri Mar 11 11:46:35.825734 2016] [authz_core:error] [pid 8719] [client 10.194.141.100:39456] AH01630: client denied by server configuration: /var/www/owncloud/data/.ocdata
[Fri Mar 11 11:51:39.376742 2016] [authz_core:error] [pid 8714] [client 10.194.141.100:39499] AH01630: client denied by server configuration: /var/www/owncloud/data/.ocdata

```

#### ownCloud log (data/owncloud.log)
```
root@owncloud:~# cat /var/www/owncloud/data/owncloud.log 
{"reqId":"A1Yf2tcWVL3j+LWWWzfQ","remoteAddr":"10.194.141.100","app":"no app in context","message":"Exception: {\"Exception\":\"DomainException\",\"Message\":\"Contacts tables are missing. Nothing to do.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/migrateaddressbooks.php(83): OCA\\\\Dav\\\\Migration\\\\AddressBookAdapter->setup()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/application.php(186): OCA\\\\Dav\\\\Migration\\\\MigrateAddressbooks->setup()\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/install.php(26): OCA\\\\Dav\\\\AppInfo\\\\Application->migrateAddressbooks()\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(621): include('\\\/var\\\/www\\\/ownclo...')\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(572): OC_Installer::includeAppScript('\\\/var\\\/www\\\/ownclo...')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(546): OC_Installer::installShippedApp('dav')\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/setup.php(370): OC_Installer::installShippedApps()\\n#7 \\\/var\\\/www\\\/owncloud\\\/core\\\/controller\\\/setupcontroller.php(64): OC\\\\Setup->install(Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(823): OC\\\\Core\\\\Controller\\\\SetupController->run(Array)\\n#9 \\\/var\\\/www\\\/owncloud\\\/index.php(39): OC::handleRequest()\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/addressbookadapter.php\",\"Line\":72}","level":3,"time":"2016-03-10T15:38:12+00:00"}
{"reqId":"A1Yf2tcWVL3j+LWWWzfQ","remoteAddr":"10.194.141.100","app":"no app in context","message":"Exception: {\"Exception\":\"DomainException\",\"Message\":\"Calendar tables are missing. Nothing to do.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/migratecalendars.php(84): OCA\\\\Dav\\\\Migration\\\\CalendarAdapter->setup()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/application.php(202): OCA\\\\Dav\\\\Migration\\\\MigrateCalendars->setup()\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/install.php(27): OCA\\\\Dav\\\\AppInfo\\\\Application->migrateCalendars()\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(621): include('\\\/var\\\/www\\\/ownclo...')\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(572): OC_Installer::includeAppScript('\\\/var\\\/www\\\/ownclo...')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(546): OC_Installer::installShippedApp('dav')\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/setup.php(370): OC_Installer::installShippedApps()\\n#7 \\\/var\\\/www\\\/owncloud\\\/core\\\/controller\\\/setupcontroller.php(64): OC\\\\Setup->install(Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(823): OC\\\\Core\\\\Controller\\\\SetupController->run(Array)\\n#9 \\\/var\\\/www\\\/owncloud\\\/index.php(39): OC::handleRequest()\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/calendaradapter.php\",\"Line\":68}","level":3,"time":"2016-03-10T15:38:12+00:00"}

```

**Comment by LukasReschke** [comment]
Will take a look next week.

**Comment by shhristov** [comment]
Thank you!

**Comment by PVince81** [comment]
@LukasReschke any update ?

**Comment by LukasReschke** [comment]
Oops. Yeah. That's fixed in 9.0.1 already, @shhristov it's indeed not yet in the packages :see_no_evil:
